SearchResults.js

```diff
@@ -0,0 +1,147 @@
+class SearchResults {
+  constructor(element) {
+    this.element = element;
+    this.data;
+  }
+
+  async dataFetch() {
+    const loadingElement = showLoadingScreen();
+    try {
+      let searchUrl = `https:stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/search?query=${input.value}&amp;limit=10&amp;exchange=NASDAQ`;
+      let searchResponse = await fetch(searchUrl);
+      let data = await searchResponse.json();
+      data;
+      loadingElement.remove();
+      return data;
+    } catch (eror) {
+      loadingElement.remove();
+      clearEror();
+      showError(eror);
+    }
+  }
+
+  getImg(data) {
+    let aTag = document.getElementById("container").querySelectorAll("p");
+    for (let i = 0; i < data.length; i++) {
+      let img = document.createElement("img");
+      img.src = data[i].profile.image;
+      img.onerror = () => {
+        img.src = "img/img.png";
+      };
+      aTag[i].appendChild(img);
+    }
+  }
+
+  getChangesPercentage(data) {
+    let aTag = document.getElementById("container").querySelectorAll(".aTag");
+    for (let i = 0; i < data.length; i++) {
+      let div = document.createElement("div");
+      div.className = "changesPercentage";
+      if (data[i].profile.changesPercentage < 0) {
+        div.innerHTML = `${data[i].profile.changesPercentage}%`;
+        div.classList.add("red");
+      } else {
+        div.innerHTML = `+${data[i].profile.changesPercentage}%`;
+        div.classList.add("green");
+      }
+      aTag[i].appendChild(div);
+    }
+  }
+
+  fetchNTimes() {
```

Comment:
this function should be inside the form class, since the searchResutls class has to handle only functionalities that realtes to parsing the response from the server and manipulating the dom